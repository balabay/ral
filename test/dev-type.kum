| Тест.  Проверка и продвижение типов

алг A
нач
	many_params(many_params(1,2,3,4), 2, 3, 4)
	сравнения()
	логические выражения()

	если 1.1 то вывод "1.1", нс всё
кон

алг цел many_params(цел a, цел b, цел c, цел d)
нач
	знач := a + b + c + d
	вывод a, ' ', b, ' ', c, ' ', d, нс
кон

алг цел сравнения()
нач
цел а = 10
вещ б = 8.9

вывод "а = ", а, " б = ", б, " а == б: ", а = б, " а != б: ", а != б, нс
вывод "а = 10.0:", а = 10.0, нс
вывод "а = 10.1:", а = 10.1, нс
вывод "(а = 10.1) = 0:", (а = 10.1) = 0, нс
вывод "б = 8:", б = 8, нс
вывод "б = 8.9:", б = 8.9, нс

вывод "а < 10:", а < 10, нс
вывод "а <= 10:", а <= 10, нс
вывод "а > 10:", а > 10, нс
вывод "а >= 10:", а >= 10, нс

вывод "б < 10:", б < 10, нс
вывод "б <= 10:", б <= 10, нс
вывод "б <= б:", б <= б, нс
вывод "б > 10:", б > 10, нс
вывод "б >= 10:", б >= 10, нс
вывод "б >= 8.9:", б >= 8.9, нс

вывод "(б < 10) == (а = 10):", (б < 10) = (а = 10), нс
вывод "б < 10 = а = 0:", б < 10 = а = 0, нс

кон

алг логические выражения()
нач
цел а = 0
вещ б = 8.9

вывод "а = ", а, " б = ", б, " а и б: ", а и б, " а или б: ", а или б, " а и б или б и а: ", а и б или б и а, " (а и б или б) или а: ", (а и б или б) или а, нс

кон
