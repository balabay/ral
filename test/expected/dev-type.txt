1 2 3 4
10 2 3 4
а = 10 б = 8.900000 а == б: 0 а != б: 1
а = 10.0:1
а = 10.1:0
(а = 10.1) = 0:1
б = 8:0
б = 8.9:1
а < 10:0
а <= 10:1
а > 10:0
а >= 10:1
б < 10:1
б <= 10:1
б <= б:1
б > 10:0
б >= 10:0
б >= 8.9:1
(б < 10) == (а = 10):1
б < 10 = а = 0:1
а = 0 б = 8.900000 а и б: 0 а или б: 1 а и б или б и а: 0 (а и б или б) или а: 1
